utilities/transactions/lock/lock_tracker.cc

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
+// This source code is licensed under both the GPLv2 (found in the
+// COPYING file in the root directory) and Apache 2.0 License
+// (found in the LICENSE.Apache file in the root directory).
+
+#include "utilities/transactions/lock/lock_tracker.h"
+
+#include "utilities/transactions/lock/point_lock_tracker.h"
+
+namespace ROCKSDB_NAMESPACE {
+
+LockTracker* NewLockTracker() {
```

Comment:
The function will need some context to know what kind of lock tracker to create  (AFAIU there's no global variable pointing which kind of lock manager is used). 
When trying to implement range locking for this, I ended up with this: https://gist.github.com/spetrunia/66c41c8a2c41d5f3f4bae27d505181fb .  Here, the Lock Manager has a function that returns a factory object that creates LockTracker objects that are suitable for use with this Lock Manager. Does this make sense?

Reply:
Yes, it makes sense.
During prototyping, originally there is a method like `bool SupportsPointLock()`, `bool SupportsRangeLock()`.
But I thought the lock manager interface can hold a LockTracker, like what you did. But I didn't do this in the current PR because I planned to do this in a future PR for new lock manager interface.

Reply:
Added `IsPointLockSupported()` and `IsRangeLockSupported()`, hope they can be useful in range lock tracker.